frontend/src/components/shared/StreamlitMarkdown/StreamlitMarkdown.tsx

```diff
@@ -180,9 +186,9 @@ function CustomParsedHtml(props: CustomParsedHtmlProps): ReactElement {
     return (ReactMarkdown.renderers.parsedHtml as any)(props)
   }
 
-  const { "data-anchor": anchor, children } = elementProps
+  const { "data-anchor": anchor, children, ...rest } = elementProps
   return (
-    <HeadingWithAnchor tag={type} anchor={anchor}>
+    <HeadingWithAnchor tag={type} anchor={anchor} tagProps={rest}>
```

Comment:
Could use a test that verifies that HTML attributes get propagated to the rendered markdown